deployment/cloud/modules/server/main.tf

```diff
@@ -0,0 +1,221 @@
+
+data "aws_caller_identity" "this" {}
+
+// vpc
+
+module "vpc" {
+    source = "terraform-aws-modules/vpc/aws"
+    version = "~> 5.0"
+
+    name = "${var.resource_prefix}-vpc"
+    cidr = "10.0.0.0/16"
+
+    azs = ["${data.aws_region.this.name}a", "${data.aws_region.this.name}b", "${data.aws_region.this.name}c"]
+    private_subnets = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24"]
+    public_subnets  = ["10.0.101.0/24", "10.0.102.0/24", "10.0.103.0/24"]
+
+    enable_nat_gateway = true
+    single_nat_gateway = true
+    one_nat_gateway_per_az = false
+    enable_dns_hostnames = true
+    enable_dhcp_options  = true
+}
+
+// instance role
+
+resource "aws_iam_role" "instance" {
+  name = "${var.resource_prefix}-instance-role"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Effect = "Allow",
+        Principal = {
+          Service = "ec2.amazonaws.com"
+        },
+        Action = "sts:AssumeRole"
+      }
+    ]
+  })
+
+  inline_policy {
+    name = "${var.resource_prefix}_instance_policy"
+    policy = jsonencode({
+      Version = "2012-10-17",
+      Statement = [
+        {
+          Effect = "Allow",
+          Action = [
+            "ec2:DescribeInstances",
+            "ec2:DescribeInstanceStatus",
+            "ec2:StartInstances",
+            "ec2:StopInstances",
+            "ec2:AssociateAddress",
+            "ec2:DisassociateAddress"
+          ],
+          Resource = "*",
+          Condition = {
+            StringEquals = {
+              "aws:ResourceTag/Project" = data.aws_default_tags.this.tags.Project
+            }
+          }
+        }
+      ]
+    })
+  }
+
+  managed_policy_arns = ["arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"]
+}
+
+
+resource "aws_iam_instance_profile" "this" {
+  name = "${var.resource_prefix}-instance-profile"
+  role = aws_iam_role.instance.name
+}
+
+// security group
+resource "aws_security_group" "this" {
+  name = "${var.resource_prefix}-security-group"
+  vpc_id = module.vpc.vpc_id
+
+# TODO: restrict to UW IPs https://github.com/uw-ssec/post-disaster-comms/issues/64
+#   ingress {
+#     from_port = 22
+#     to_port = 22
+#     protocol = "tcp"
+#     cidr_blocks = []
+#   }
+
+  ingress {
+    from_port = 443
+    to_port = 443
+    protocol = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    from_port = 80
+    to_port = 80
+    protocol = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+// launch template -- userdata tbd for now
+
+resource "aws_launch_template" "this" {
+  name = "${var.resource_prefix}-template"
+  // https://documentation.ubuntu.com/aws/en/latest/aws-how-to/instances/find-ubuntu-images/#finding-images-with-ssm
+  image_id = "resolve:ssm:/aws/service/canonical/ubuntu/server/jammy/stable/current/amd64/hvm/ebs-gp2/ami-id"
+  instance_type = "r5.large"
```

Comment:
Should we make this a variable so that we can specify the instance type?

Reply:
i don't see why not! i'll have it default to `r5.large` for now